### ISSUE TITLE
Declaring reusable variables

### ISSUE THREAD

**Issue by OskarPersson** [kaj/ructe#78]
Is there something similar to Django's [`with`](https://docs.djangoproject.com/en/3.0/ref/templates/builtins/#with) tag to store result of complex or expensive methods for use in the template multiple times?

**Comment by kaj** [comment]
Not really, unles you count `@if let ...`, `@match ...`, or function / template calls.

It has been suggested before, so maybe it is a good idea.  I guess there could be an `@let` template tag similar to `@if let`, but with an irrefutable pattern (you can actually use `@if let variable = expression { ... }`, but rustc will give you a warning about the "irrefutable if-let pattern").
